app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -86,109 +87,119 @@ fun NiaApp(
         windowSizeClass = windowSizeClass
     ),
 ) {
-    val background: @Composable (@Composable () -> Unit) -> Unit =
-        when (appState.currentTopLevelDestination) {
-            TopLevelDestination.FOR_YOU -> {
-                content ->
-                NiaGradientBackground(content = content)
-            }
-            else -> { content -> NiaBackground(content = content) }
-        }
+    val showGradientBackground = appState.currentTopLevelDestination == TopLevelDestination.FOR_YOU
 
-    background {
+    NiaBackground {
+        NiaGradientBackground(
+            topColor = if (showGradientBackground) {
```

Comment:
I've refactored this in #502

Reply:
Sounds good, do you want to merge that PR into this one before merging this one, or merge #502 after this is merged?

Reply:
Merge into this one - makes sense to keep it as a single PR

Reply:
Added your commit to this branch!